bson/_cbsonmodule.c

```diff
@@ -1747,6 +1747,9 @@ int write_dict(PyObject* self, buffer_t buffer,
                     PyObject *etype = NULL, *evalue = NULL, *etrace = NULL;
                     PyErr_Fetch(&etype, &evalue, &etrace);
                     PyObject *InvalidDocument = _error("InvalidDocument");
+                    if (InvalidDocument == NULL) {
+                        return 0;
```

Comment:
This return case will leak `&etype, &evalue, &etrace`. We either need to call PyErr_Restore or xdecref those explicitly.

Reply:
Fixed